format/spec.md

```diff
@@ -1308,7 +1321,7 @@ Default values are added to struct fields in v3.
 
 Types `timestamp_ns` and `timestamptz_ns` are added in v3.
 
-All readers are required to read tables with unknown partition transforms, ignoring them.
+All readers are required to read tables with unknown partition transforms, ignoring the unsupported partition fields.
```

Comment:
we might want to clarify this meaning in the case of matching delete files with data files, the partition values can't be ignored we just need to use the empirical schema for the types from the manifest files?  I'm also not sure if there is an impact of partition statistics (reading quickly it doesn't look like there is any)?

Reply:
I guess that we do still need to consider partition equality. I'll update.

Reply:
I updated the section on delete file matching to have a note about this as it affects partition equality.